### PR TITLE
feat(editor): migrated to nx

### PR DESCRIPTION
**Pull request by Itrulia**
`packages/editor` is now in `apps/editor`

1. Codegen is now for the whole project
2. All tests now have a .test postfix in their filename
3. runServer is gone, is now just calling the main.js
4. Future will just use .env.local

### CODE REVIEW

File: apps/editor/server.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+const app = express()
+
+const browserDist = path.join(process.cwd(), 'dist/apps/editor/browser')
+const indexPath = path.join(browserDist, 'index.html')
+
+app.use(cors())
+
+app.get('*.*', express.static(browserDist, {}))
+
+app.use('*', handleRequest(indexPath))
```
[the remark this answers is not in the document]
Comment (Itrulia) [reply apps/editor/server.ts:19]: False positive

File: apps/editor/src/app/utility.ts [hunk 2]
```diff
@@ -202,10 +205,10 @@
   if (url) {
     const pattern = new RegExp(
       '^(https?:\\/\\/)?' + // protocol
-      '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' + // domain name
-      '((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
-      '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
-      '(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
+        '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' + // domain name
```
[the remark this answers is not in the document]
Comment (Itrulia) [reply apps/editor/src/app/utility.ts:208]: Already existed before

File: apps/editor/src/app/atoms/loginTemplate.tsx [hunk 3]
```diff
@@ -18,7 +18,8 @@ export function LoginTemplate({backgroundChildren, children}: LoginTemplateProps
         backgroundColor: 'white',
 
         width: '100%',
-        height: '100%'
+        height: '100%',
+        minHeight: '100vh'
```
Comment (Itrulia) [inline apps/editor/src/app/atoms/loginTemplate.tsx:22]: fixes height problem that started because doctype was added to html

File: Dockerfile [hunk 4]
```diff
@@ -7,23 +7,21 @@ RUN mkdir -p /home/node/wepublish
...
+COPY --chown=node:node ./tsconfig.base.json ./tsconfig.base.json
+COPY --chown=node:node ./babel.config.json ./babel.config.json
 COPY --chown=node:node ./LICENSE ./LICENSE
 
 COPY --chown=node:node ./packages/ ./packages/
+COPY --chown=node:node ./apps/ ./apps/
+COPY --chown=node:node ./examples/ ./examples/
 
 RUN yarn install
+RUN yarn build:demo
```
Comment (Itrulia) [inline Dockerfile:22]: Multi staging removed

File: apps/editor/src/app/app.tsx [hunk 5]
```diff
@@ -448,5 +448,3 @@ export function App() {
     </CustomProvider>
   )
 }
-
-export const HotApp = hot(App)
```
Comment (Itrulia) [inline apps/editor/src/app/app.tsx:452]: Hot reload comes through Nx now

File: apps/editor/src/app/atoms/searchAndFilter/userSearch.tsx [hunk 6]
```diff
@@ -106,17 +106,16 @@ export function UserSearch({
           key={`user-id-${resetFilterKey}`}
           placeholder={placeholder}
           block
+          name={name}
           disabled={loading || !!error}
-          data={users.map(usr => ({value: usr?.id, label: getUserLabel(usr)}))}
```
Comment (Itrulia) [inline apps/editor/src/app/atoms/searchAndFilter/userSearch.tsx:110]: TODO: figure out why this change happened

File: apps/editor/src/app/global.less [hunk 7]
```diff
@@ -119,43 +119,4 @@
     top: 0;
     background-color: #1675e0;
   }
-}
-
-.react-icons {
```
Comment (Itrulia) [inline apps/editor/src/app/global.less:124]: Changes were needed because with switch of icon library we lost the rs-icon class. Readded it to the icons so these changes are not needed anymore

File: apps/editor/src/app/atoms/searchAndFilter/userSearch.tsx (hunk 6 above)
Comment (Itrulia) [reply apps/editor/src/app/atoms/searchAndFilter/userSearch.tsx:110]: ah, formData already is `users.map(usr => ({value: usr?.id, label: getUserLabel(usr)}))`

File: apps/editor/src/app/atoms/modelTitle.tsx [hunk 8]
```diff
@@ -68,18 +68,13 @@ export function ModelTitle({
...
-              <h1>
-                <MdChevronLeft />
-              </h1>
-            </Link>
-          </Col>
-          <Col xs={16}>
-            <h2>{titleView()}</h2>
-          </Col>
-        </Row>
+        <FlexboxGrid align="middle">
```
Comment (Itrulia) [inline apps/editor/src/app/atoms/modelTitle.tsx:71]: Fixes proper alignment, a11y & html validity (as a chevron was an h1 instead of the title)

File: apps/editor/src/app/atoms/placeholderInput.tsx [hunk 9]
```diff
@@ -23,7 +24,8 @@ export function PlaceholderInput({
   children,
   onAddClick,
   disabled,
-  maxHeight = 450
+  maxHeight = 450,
+  minHeight
```
Comment (Itrulia) [inline apps/editor/src/app/atoms/placeholderInput.tsx:28]: needed for some places where there's no specified height. So height: 100% did nothing.

**Review by tomaszdurka — APPROVED** [review]
